### ISSUE TITLE
Including ApiController attribute results on an Unsupported Media Type

### ISSUE THREAD

**Issue by zinov** [dotnet/core#5328]
Web Api project, .net core version: 3.1

My API didn't have the ApiController attribute and I was including it as part of the new changes to integrate later with swagger.
**Problem**:
All my get request were working before normally with my Endpoint. After including the attribute I got an UnsupportedMediaType error.

My original API controller looks like this:

`    [Authorize]
    [Route("api/tenants/{tenantId}/policies")]
    [EnableQuery]
    public class PolicyController : BaseController
    {
          ....
    }
`
my new change: 

`    [Authorize]
     [ApiController]
    [Route("api/tenants/{tenantId}/policies")]
    [EnableQuery]
    public class PolicyController : BaseController
    {
          ....
    }
`
my Action method:

`[HttpGet(Name = nameof(GetPoliciesByTenant))]
        public async Task<IActionResult> GetPoliciesByTenant(Guid tenantId, PolicyResourceParameterDto polResParamDto)
        {
            ....
        }
`

From the client, I was passing an **Accept** header with **application/json** which is one of the default formatters that .net core has.  I think this is a bug. Please let me know if you need more details.

**Comment by riddletom76** [comment]
@zinov in the Get request from the client are you explicitly sending a body, usually by convention the Get requests do not have a body the content is read from the URI, but in case you want to send something explicitly from the request body you need to specify in the Method in as to read it from Body.

**Comment by zinov** [comment]
No, GET request should not pass a body. Semantically it doesn't make sense, even if the http protocol allows it, the body should be ignored. Just POST, PUT and PATCH should use the body, but in that case is a Content-Type header what you need to pass.
The Dto I am passing here binds the parameters from the querystring by default. But for this example is not relevant, you can remove it and still you are gonna get the same error.
